### ISSUE TITLE
`Request::set_context` recurs infinitely

### ISSUE THREAD

**Issue by skyzh** [tikv/client-rust#340]
https://github.com/tikv/client-rust/blob/74db41c3e5ed1899708215fc284a4673f179f6e7/tikv-client-store/src/request.rs#L41-L43

Is this expected... or a bug?

**Comment by skyzh** [comment]
In https://github.com/tikv/client-rust/pull/339, I've set it to a no-op.

**Comment by skyzh** [comment]
After some investigation, I think clippy is producing false warnings...

```
2022-03-20T04:52:00.0853586Z ##[error]Unable to create clippy annotations! Reason: HttpError: Resource not accessible by integration
2022-03-20T04:52:00.0864655Z ##[warning]It seems that this Action is executed from the forked repository.
2022-03-20T04:52:00.0866821Z ##[warning]GitHub Actions are not allowed to create Check annotations, when executed for a forked repos. See https://github.com/actions-rs/clippy-check/issues/2 for details.
2022-03-20T04:52:00.0867664Z Posting clippy checks here instead.
2022-03-20T04:52:00.0867995Z error: parameter is only used in recursion
2022-03-20T04:52:00.0868432Z   --> tikv-client-store/src/request.rs:41:39
2022-03-20T04:52:00.0869062Z    |
2022-03-20T04:52:00.0869379Z 41 |             fn set_context(&mut self, context: kvrpcpb::Context) {
2022-03-20T04:52:00.0869819Z    |                                       ^^^^^^^ help: if this is intentional, prefix with an underscore: `_context`
2022-03-20T04:52:00.0870343Z ...
2022-03-20T04:52:00.0870647Z 48 | impl_request!(RawGetRequest, raw_get_async_opt, "raw_get");
2022-03-20T04:52:00.0871224Z    | ---------------------------------------------------------- in this macro invocation
2022-03-20T04:52:00.0871567Z    |
2022-03-20T04:52:00.0872069Z    = note: `-D clippy::only-used-in-recursion` implied by `-D clippy::all`
2022-03-20T04:52:00.0872769Z    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#only_used_in_recursion
2022-03-20T04:52:00.0873516Z    = note: this error originates in the macro `impl_request` (in Nightly builds, run with -Z macro-backtrace for more info)
2022-03-20T04:52:00.0873821Z 
```

The kvrpcpb structs have some wrapper files in generated protos for `set_context`.

To avoid confusion, I've modified this to fully-quantified syntax:

```
kvrpcpb::$name::set_context(self, context)
```
